stl/inc/flat_map

```diff
@@ -1206,6 +1183,24 @@ private:
         }
     }
 
+    template <class _KeyTy, class... _MappedArgTypes>
+    pair<iterator, bool> _Try_emplace(_KeyTy&& _Key_val, _MappedArgTypes&&... _Mapped_args) {
+        auto _Key_it = _STD lower_bound(_Data.keys.begin(), _Data.keys.end(), _Key_val, _Key_compare);
+        if (_Key_it != _Data.keys.end() && _Key_equal(*_Key_it, _STD forward<_KeyTy>(_Key_val))) {

```

Comment:
I think there's a widespread performance issue around uses of `_Key_equal`. It performs two comparisons, but here it's being called after `lower_bound`, so we should need only one more comparison.

Unlike many micro-optimization issues, key comparisons can be very expensive.